app/Http/Controllers/BookingRequestController.php

```diff
@@ -154,4 +156,25 @@ public function destroy(BookingRequest $booking)
 
         return redirect(route('bookings.index'));
     }
+
+    public function downloadReferenceFiles($folder) 
+    {
+        $path = Storage::disk('public')->path($folder);
+
+        $zip = new ZipArchive;
+   
+        $fileName = $folder . '.zip';
+   
+        $zip->open(Storage::disk('public')->path($fileName), ZipArchive::CREATE);
+        $files = File::files($path);
+                              
+        foreach ($files as $file) {
+            $relativeNameInZipFile = basename($file);
+            $zip->addFile($file, $relativeNameInZipFile);
+        }             
+        $zip->close();
+
+        return response()->download(Storage::disk('public')->path($fileName))->deleteFileAfterSend(true);
```

Comment:
Does deleteFileAfterSend remove the file entirely from the system? Is that something that should be done, since these are reference files for bookings that may be needed later?

Reply:
It does not, all it does is removes the zip file that was downloaded, the original folder that contains the reference documents is untouched.

Reply:
Alright, thanks. Once the branch is up to date with main I'll approve. If we add other buttons to the booking list, it'll be in a different feature.